src/helpers/paper-util.js

```diff
@@ -297,9 +294,28 @@ const PaperUtil = {
       this.groups[group].addChild(circle)
     }
 
-    this.createLabel = (group: GroupKeyT, centerGridPoint: PointT, text: string, anchor: AnchorT = 'left') => {
-      const paperCenterPoint = this.fromGridCoordinateToView(centerGridPoint)
-      const label = createLabel(paperCenterPoint, text, anchor)
+    this.createLabel = (group: GroupKeyT, axisBoundsGridCoordinatePoint: PointT, text: string, anchor: AnchorT = 'left') => {
+      const axisBoundsPoint = this.fromGridCoordinateToView(axisBoundsGridCoordinatePoint)
+      let nextClosestVerticalPoint = null
+      if (anchor === 'bottom') {
+        nextClosestVerticalPoint = this.fromGridCoordinateToView({ x: axisBoundsGridCoordinatePoint.x, y: axisBoundsGridCoordinatePoint.y + 1 })
+      } else {
+        nextClosestVerticalPoint = this.fromGridCoordinateToView({ x: axisBoundsGridCoordinatePoint.x, y: axisBoundsGridCoordinatePoint.y + 1 })
```

Comment:
what the difference between the position above?

Reply:
This ended up making no difference because the point is in bounds in this case, but technically to ensure it's in bounds we should be moving up from the bottom and down from everywhere else, fixed this in the next commit.